File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */

var strStr = function(haystack, needle) {
    let hLen = haystack.length, nLen = needle.length;
    if (nLen === 0) return 0;

    for (let i = 0; i <= hLen - nLen; i++) {
        if (haystack.substring(i, i + nLen) === needle) {
            return i;
        }
    }

    return -1;
};

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
  int f=-1;
  int j=0;
        for(int i=0;i<needle.length();i++){
            if(haystack.charAt(i)!=needle.charAt(j)){
                 j++;
                
                
               
            }
         
        f=haystack.indexOf(needle);
        return f;
            
        }
        return f;
    }
}